src/fields/time/time-helper.ts

```diff
@@ -0,0 +1,45 @@
+export namespace TimeHelper {
+	export const validate = (value: string, is24HourFormat = false) => {
+		const timeArr = value.split(":");
+
+		if (!is24HourFormat) {
+			// Check format
+			if (timeArr.length !== 2 || timeArr[1].length !== 4) return false;
+
+			const minute = timeArr[1].substring(0, 2);
+			const period = timeArr[1].substring(2);
+
+			if (!isValidHour(timeArr[0], is24HourFormat) || !isValidMinutes(minute) || !isValidTimePeriod(period)) {
+				return false;
+			}
+
+			return true;
+		} else {
+			// Check format
+			if (timeArr.length !== 2 || timeArr[1].length === 4) return false;
+
+			if (!isValidHour(timeArr[0], is24HourFormat) || !isValidMinutes(timeArr[1])) {
+				return false;
+			}
+
+			return true;
+		}
+	};
+}
+
+// =============================================================================
+// NON-EXPORTABLES
+// =============================================================================
+const isValidHour = (hourString: string, is24HourFormat = false): boolean => {
+	const numValue = parseInt(hourString);
```

Comment:
should use `Number(hourString)` instead as `parseInt("1a")` can still return a number part, meaning `1a:3bpm` passes validation

Reply:
but this will now need `Number(hourString).isInteger()` also to check that it's an integer otherwise `.1:20pm` will pass

Reply:
ok I updated the check to be `Number(hourString)===parseInt(hourString)`